tests/utils/test_fmrisim.py

```diff
@@ -138,6 +231,73 @@ def test_apply_signal():
                                        tr_duration=tr_duration,
                                        )
 
+    # Check that you can compute signal change appropriately
+    # Preset a bunch of things
+    stimfunction_tr = stimfunction[::int(tr_duration * 100)]
+    mask, template = sim.mask_brain(dimensions, mask_self=False)
+    noise_dict = sim._noise_dict_update({})
+    noise = sim.generate_noise(dimensions=dimensions,
+                               stimfunction_tr=stimfunction_tr,
+                               tr_duration=tr_duration,
+                               template=template,
+                               mask=mask,
+                               noise_dict=noise_dict,
+                               iterations=[0, 0]
+                               )
+    coords = feature_coordinates[0]
+    noise_function = noise[coords[0], coords[1], coords[2], :]
+    noise_function = noise_function.reshape(duration // tr_duration, 1)
+
+    # Create the calibrated signal with PSC
+    magnitude = [0.5]
+    sig_a = sim.compute_signal_change(signal_function,
+                                      noise_function,
+                                      noise_dict,
+                                      magnitude,
+                                      'PSC',
+                                      )
+    magnitude = [1]
+    sig_b = sim.compute_signal_change(signal_function,
+                                      noise_function,
+                                      noise_dict,
+                                      magnitude,
+                                      'PSC',
+                                      )
+
+    assert (abs(sig_b) - abs(sig_a)).min() >= 0, 'Magnitude modulation failed'
```

Comment:
If the noise_function and signal_function are shared between sig_a and sig_b, is it expected that sig_b is twice as big as sig_a for all time points and voxels? If so, maybe this can be tested as well.
And I think it might be worth testing the result of the other metrics as well, in addition to verifying they can run.

Reply:
Good suggestion, I added these tests, it is a good safety precaution